### PR TITLE
Fix Sonic-config-engine testcase

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity <sangitamaity0211@gmail.com>

**- What I did**
fixed **test_cfggen_platformJson.py**  test script as it was broken after introduction of **hwsku.json**

**- How I did it**

1. Added a sample **hwsku.json** file in **"src/sonic-config-engine/tests"** folder to use during test.
2. Added new option **"-S"/"--hwsku-config"** to **sonic-cfggen** as it is necessary to have some kind of default file during image build up especially for test cases to pass.
3. using new option **"-S"/"--hwsku-config"** to test file to get results.

**- How to verify it**
**sonic_config_engine-1.0-py2-none-any.whl** package got built successfully with all test cases passed.
```
samaity@server09:~/REPO_FACTORY/GIT_REPO/DPB/sonic-buildimage$ BLDENV=stretch make target/python-wheels/sonic_config_engine-1.0-py2-none-any.whl
+++ --- Making target/python-wheels/sonic_config_engine-1.0-py2-none-any.whl --- +++
EXTRA_JESSIE_TARGETS=sonic_config_engine-1.0-py2-none-any.whl make -f Makefile.work jessie
make[1]: Entering directory '/home/samaity/REPO_FACTORY/GIT_REPO/DPB/sonic-buildimage'
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "broadcom"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20200214.092507"
"BLDENV"                          : "stretch"
"VS_PREPARE_MEM"                  : "yes"
"ENABLE_SFLOW"                    : "y"

make: Nothing to be done for 'jessie'.
make[1]: Leaving directory '/home/samaity/REPO_FACTORY/GIT_REPO/DPB/sonic-buildimage'
BLDENV=stretch make -f Makefile.work target/python-wheels/sonic_config_engine-1.0-py2-none-any.whl
make[1]: Entering directory '/home/samaity/REPO_FACTORY/GIT_REPO/DPB/sonic-buildimage'
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "broadcom"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20200214.092514"
"BLDENV"                          : "stretch"
"VS_PREPARE_MEM"                  : "yes"
"ENABLE_SFLOW"                    : "y"

make[1]: Leaving directory '/home/samaity/REPO_FACTORY/GIT_REPO/DPB/sonic-buildimage'

```

```
samaity@server09:~/REPO_FACTORY/GIT_REPO/DPB/sonic-buildimage$ ls -la target/python-wheels/sonic_config_engine-1.0-py2-none-any.whl
-rw-r--r-- 1 samaity samaity 45343 Feb 14 09:25 target/python-wheels/sonic_config_engine-1.0-py2-none-any.whl
```

### CODE REVIEW

File: src/sonic-config-engine/tests/test_cfggen_platformJson.py [hunk 1]
```diff
@@ -42,27 +42,28 @@ def test_print_data(self):
 
     # Check whether all interfaces present or not as per platform.json
     def test_platform_json_interfaces_keys(self):
-        argument = '-m "' + self.sample_graph_simple + '" -p "' + self.platform_json + '" -v "PORT.keys()"'
+        argument = '-m "' + self.sample_graph_simple + '" -p "' + self.platform_json + '" -S "' + self.hwsku_json + '" -v "PORT.keys()"'
         output = self.run_script(argument)
+        print("kkkkkkk -> {}".format(output))
```
Comment (zhenggen-xu) [inline src/sonic-config-engine/tests/test_cfggen_platformJson.py]: Is this test code?

File: src/sonic-config-engine/portconfig.py [hunk 2]
```diff
@@ -114,9 +114,15 @@ def get_port_config(hwsku=None, platform=None, port_config_file=None):
         if not port_config_file:
             return ({}, {})
 
+
     # Read from 'platform.json' file
     if port_config_file.endswith('.json'):
-        hwsku_json_file = get_hwsku_file_name(hwsku, platform)
+        if not hwsku_config_file:
+             hwsku_json_file = get_hwsku_file_name(hwsku, platform)
+             if not hwsku_config_file:
```
Comment (zhenggen-xu) [inline src/sonic-config-engine/portconfig.py]: First, this should be "if not hwsku_json_file" ?

The behavior is changed here.  If we don't specify hwsku.json and it does not exist,  should we return empty or raise?  Not sure how many caller use this function, can they handle empty return? I think we should  raise.

File: src/sonic-config-engine/portconfig.py [hunk 3]
```diff
@@ -114,9 +114,15 @@ def get_port_config(hwsku=None, platform=None, port_config_file=None):
...
+                return ({}, {})
+        else:
+            hwsku_json_file = hwsku_config_file
         if not hwsku_json_file:
```
Comment (zhenggen-xu) [inline src/sonic-config-engine/portconfig.py]: This is never hit after the change.

File: src/sonic-config-engine/tests/test_cfggen_platformJson.py (hunk 1 above)
Comment (samaity) [reply src/sonic-config-engine/tests/test_cfggen_platformJson.py]: yes, you are right. Removed the line.

File: src/sonic-config-engine/portconfig.py (hunk 2 above)
Comment (samaity) [reply src/sonic-config-engine/portconfig.py]: Modified the line to "if not hwsku_json_file". 
If we don't specify hwsku.json and it does not exist, we should return empty as it's the expected behavior of **"get_port_config"** from beginning. Whoever calls  "get_port_config" function, should handle an empty return. That's the current behavior of code. 

Let me know if I am missing something.

File: src/sonic-config-engine/portconfig.py (hunk 3 above)
Comment (samaity) [reply src/sonic-config-engine/portconfig.py]: yes, I have removed the check.

**Review by zhenggen-xu — APPROVED** [review]

**Pull request merged**